### PR TITLE
fix: added self-contained styles for list-group

### LINKED ISSUE #212: Self-contained style for list-group component
Refactoring to make the list-group component support self-contained style.
This PR is part of the larger task of making all components self-contained. #136

### PR DESCRIPTION
**Pull request by droshev**
fixes #212 

Screen Shot 2019-08-14 at 8 28 02 AM

### CODE REVIEW

File: test/templates/list-group/index.njk [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
 {% from "./component.njk" import list_group %}
-{% from "../form/component.njk" import form_item %}
-{% from "../button/component.njk" import button %}
-{% from "../icon/component.njk" import icon %}
 {% from "./../format.njk" import format %}
 
 {% block content %}
+<style>
+
+.test-pg__checkbox{
```
Comment (jbadan) [inline test/templates/list-group/index.njk]: This looks great! 

@greg-a-smith just merged a pr with a shared-layout stylesheet for the placeholder styles, can we move this in there?

 https://github.com/SAP/fundamental-styles/blob/master/test/resources/layout-shared.css

Comment (droshev) [reply test/templates/list-group/index.njk]: sure!

File: components/list-group.scss [hunk 2]
```diff
@@ -0,0 +1 @@
+@import "../scss/components/list-group";
```
Comment (jbadan) [inline components/list-group.scss]: This file needs fonts, then I think we're good to go!

File: test/resources/layout-shared.css [hunk 3]
```diff
@@ -10,3 +10,10 @@ body {
     background: beige;
     font-size: 14px;
 }
+
+.test-pg__checkbox {
+    width: 15px;
+    height: 15px;
+    border: 1px solid magenta;
+    border-radius: 2px;
+}
```
Comment (greg-a-smith) [inline test/resources/layout-shared.css]: Looks like we can just get rid of the checkbox examples as they have no impact on the list group styles.

File: test/templates/list-group/component.njk [hunk 4]
```diff
@@ -1,23 +1,12 @@
 {% import "../forms.njk" as forms %}
-{% from "../button/component.njk" import button %}
 {% from "../icon/component.njk" import icon %}
```
Comment (greg-a-smith) [inline test/templates/list-group/component.njk]: We shouldn't be importing the icon component.

File: test/templates/list-group/index.njk [hunk 5]
```diff
@@ -13,19 +10,19 @@
...
 {{ format(example) }}
 
-<br><br>
+<br /><br />
 <h2>with checkboxes</h2>
 {% set example %}
 {{  list_group(
       properties={
-          items: [form_item("checkbox",{ label: "List item 1" }),form_item("checkbox",{ label: "List item 2 sit amet consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua" }),form_item("checkbox",{ label: "List item 3" })]
+          items: ["<div class='test-pg__checkbox'></div>I have a bike","<div class='test-pg__checkbox'></div>I have a motobike","<div class='test-pg__checkbox'></div>I have a car"]
```
Comment (greg-a-smith) [inline test/templates/list-group/index.njk]: Looks like we can just get rid of the checkbox examples as they have no impact on the list group styles.

File: test/templates/list-group/component.njk (hunk 4 above)
Comment (droshev) [reply test/templates/list-group/component.njk]: I think I removed it

File: components/list-group.scss (hunk 2 above)
Comment (droshev) [reply components/list-group.scss]: done

File: test/templates/list-group/component.njk (hunk 4 above)
Comment (jbadan) [reply test/templates/list-group/component.njk]: @droshev once you remove this import statement I think this is good to 🚢 !

**Comment by hertweckhr1** [comment]
I know this is already closed, but topic for consideration - why doesn't the playground have examples of list group with checkboxes for self contained? 

image

**Comment by greg-a-smith** [comment]
@hertweckhr1 They were removed because there was no styling information for checkboxes in the `fd-list-group` classes.  All the checkbox styling is tied to `fd-form`-related classes.  It was left in the docs pages as an example of using these in a real use case.
```html
<ul class="fd-list-group">
    <li class="fd-list-group__item">
        <div class="fd-form__item fd-form__item--check">
            <label class="fd-form__label" for="121Sd939">
                <input type="checkbox" class="fd-form__control" id="121Sd939">
                List item 1
            </label>
        </div>
    </li>
    <li class="fd-list-group__item">
        <div class="fd-form__item fd-form__item--check">
            <label class="fd-form__label" for="CndSd399">
                <input type="checkbox" class="fd-form__control" id="CndSd399">
                List item 2 sit amet consectetur adipisicing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
            </label>
        </div>
    </li>
    <li class="fd-list-group__item">
        <div class="fd-form__item fd-form__item--check">
            <label class="fd-form__label" for="S8jyH416">
                <input type="checkbox" class="fd-form__control" id="S8jyH416">
                List item 3
            </label>
        </div>
    </li>
</ul>
```

**Pull request merged**